da/celestia/celestia.go

```diff
@@ -111,60 +127,101 @@ func (c *DataAvailabilityLayerClient) SubmitBlock(ctx context.Context, block *ty
 }
 
 // CheckBlockAvailability queries DA layer to check data availability of block at given height.
-func (c *DataAvailabilityLayerClient) CheckBlockAvailability(
-	ctx context.Context, dataLayerHeight uint64,
-) da.ResultCheckBlock {
-	shares, err := c.client.NamespacedShares(ctx, c.namespaceID, dataLayerHeight)
-	code := dataRequestErrorToStatus(err)
-	if code != da.StatusSuccess {
+func (c *DataAvailabilityLayerClient) CheckBlockAvailability(ctx context.Context, dataLayerHeight uint64) da.ResultCheckBlock {
+	header, err := c.rpc.Header.GetByHeight(ctx, dataLayerHeight)
+	if err != nil {
+		return da.ResultCheckBlock{
+			BaseResult: da.BaseResult{
+				Code:    da.StatusError,
+				Message: err.Error(),
+			},
+		}
+	}
+	if header.DAH == nil {
+		return da.ResultCheckBlock{
+			BaseResult: da.BaseResult{
+				Code:     da.StatusSuccess,
+				DAHeight: dataLayerHeight,
+			},
+			DataAvailable: false,
+		}
+	}
+	err = c.rpc.Share.SharesAvailable(ctx, header.DAH)
+	if err != nil {
+		if err.Error() == ErrNotAvailable.Error() {
+			return da.ResultCheckBlock{
+				BaseResult: da.BaseResult{
+					Code:     da.StatusSuccess,
+					DAHeight: dataLayerHeight,
+				},
+				DataAvailable: false,
+			}
+		}
 		return da.ResultCheckBlock{
 			BaseResult: da.BaseResult{
 				Code:    da.StatusError,
 				Message: err.Error(),
 			},
 		}
 	}
-
 	return da.ResultCheckBlock{
 		BaseResult: da.BaseResult{
 			Code:     da.StatusSuccess,
 			DAHeight: dataLayerHeight,
 		},
-		DataAvailable: len(shares) > 0,
+		DataAvailable: true,
 	}
 }
 
 // RetrieveBlocks gets a batch of blocks from DA layer.
-func (c *DataAvailabilityLayerClient) RetrieveBlocks(
-	ctx context.Context, dataLayerHeight uint64,
-) da.ResultRetrieveBlocks {
-	data, err := c.client.NamespacedData(ctx, c.namespaceID, dataLayerHeight)
-	status := dataRequestErrorToStatus(err)
-	if status != da.StatusSuccess {
+func (c *DataAvailabilityLayerClient) RetrieveBlocks(ctx context.Context, dataLayerHeight uint64) da.ResultRetrieveBlocks {
+	header, err := c.rpc.Header.GetByHeight(ctx, dataLayerHeight)
+	if err != nil {
```

Comment:
Unlike earlier, we need to pass. the `DataAvailabilityHeader` of the height at which the block was submitted, so this needs to be reworked.

Reply:
This is fixed by using `blob.GetAll` instead.